internal/services/manageddevopspools/schema_definitions.go

```diff
@@ -0,0 +1,469 @@
+package manageddevopspools
+
+import (
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devopsinfrastructure/2025-01-21/pools"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+func AgentProfileSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Required: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"kind": {
+					Type:     pluginsdk.TypeString,
+					Required: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						string("Stateless"),
+						string("Stateful"),
+					}, false),
+				},
+				"grace_period_time_span": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+				"max_agent_lifetime": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+				"resource_predictions":         ResourcePredictionsSchema(),
+				"resource_predictions_profile": ResourcePredictionsProfileSchema(),
+			},
+		},
+	}
+}
+
+func ResourcePredictionsSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"time_zone": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+				},
+				"days_data": {
```

Comment:
From its doc, this field only accepts `json` format, maybe add a JSON string validation function?